src/main/java/net/reini/rabbitmq/cdi/ConnectionConfiguration.java

```diff
@@ -8,29 +13,43 @@
 import java.util.Set;
 import java.util.concurrent.TimeoutException;
 
-import javax.net.ssl.SSLContext;
-
-import com.rabbitmq.client.Address;
-import com.rabbitmq.client.Connection;
-import com.rabbitmq.client.ConnectionFactory;
-
 /**
  * Contains the Rabbit MQ connection configuration
  *
  * @author Patrick Reinhart
  */
-class ConnectionConfiguration implements ConnectionConfig, ConnectionConfigHolder {
+public class ConnectionConfiguration implements ConnectionConfigHolder {
+
+  private static final int DEFAULT_CONNECTION_HEARTBEAT_TIMEOUT_IN_SEC = 3;
+  private static final int DEFAULT_CONNECT_TIMEOUT_IN_MS = 10000;
+  private static final int DEFAULT_WAIT_TIME_RETRY_CONNECT_IN_MS = 10_000;
+  private static final long DEFAULT_WAIT_TIME_RETRY_ACTIVATE_CONSUMER_IN_MS = 10000;
+
   private final List<Address> brokerHosts;
 
   private boolean secure;
   private String username;
   private String password;
   private String virtualHost;
-
-  ConnectionConfiguration() {
+  private int requestedConnectionHeartbeatTimeout;
+  private int connectTimeout;
+  private long connectRetryWaitTime;
+  private long failedConsumerActivationRetryTime;
+  private SSLContextFactory sslContextFactory;
+
+  public ConnectionConfiguration(SSLContextFactory sslContextFactory) {
```

Comment:
Public access relay needed?

Reply:
Totally agree it should be package privat if not needed i check that

Reply:
fixed